src/router/payload-encoder.ts

```diff
@@ -12,6 +12,23 @@ import { SwapSide } from '../constants';
 import { DexAdapterService } from '../dex';
 import { convertToBasisPoints } from '../utils';
 
+export function encodeFeePercent(
+  partnerFeePercent: string,
+  positiveSlippageToUser: boolean,
+) {
+  let fee = BigInt(partnerFeePercent);
+  if (fee > 10000) throw new Error('fee bps should be less than 10000');
+
+  // Set 14th bit if positiveSlippageToUser is true
+  if (positiveSlippageToUser) fee = fee | (BigInt(1) << BigInt(14));
```

Comment:
I'd recommend calculating (BigInt(1) << BigInt(14) once and the using the value as much as needed

Reply:
fixed